migration/v2.21.0-ck8s1-v2.21.0-ck8s2/upgrade-cluster.md

```diff
@@ -0,0 +1,70 @@
+# Upgrade v2.21.0-ck8s1 to v2.21.0-ck8s2
+
+## Prerequisites
+
+- [ ] Notify the users (if any) before the upgrade starts;
+- [ ] Check if there are any pending changes to the environment;
+- [ ] Check the state of the environment, pods, nodes and backup jobs:
+
+    ```bash
+    ./compliantkubernetes-apps/bin/ck8s test sc|wc
+    ./compliantkubernetes-apps/bin/ck8s ops kubectl sc|wc get pods -A -o custom-columns=NAMESPACE:metadata.namespace,POD:metadata.name,READY-false:status.containerStatuses[*].ready,REASON:status.containerStatuses[*].state.terminated.reason | grep false | grep -v Completed
+    ./compliantkubernetes-apps/bin/ck8s ops kubectl sc|wc get nodes
+    ./compliantkubernetes-apps/bin/ck8s ops kubectl sc|wc get jobs -A
+    velero get backup
+    ```
+
+- [ ] Silence the notifications for the alerts. e.g you can use [alertmanager silences](https://prometheus.io/docs/alerting/latest/alertmanager/#silences);
+
+## Steps that can be done before the upgrade - non-disruptive
+
+1. Checkout the new release: `git switch -d v2.21.0-ck8s2`
+
+1. Switch to the correct remote: `git submodule sync`
+
+1. Update the kubespray submodule: `git submodule update --init --recursive`
+
+1. Download the required files on the nodes
+
+    ```bash
+    ./bin/ck8s-kubespray run-playbook sc upgrade-cluster.yml -b --tags=download
+    ./bin/ck8s-kubespray run-playbook wc upgrade-cluster.yml -b --tags=download
+    ```
+
+## Upgrade steps
+
+These steps will cause disruptions in the environment.
+
+1. Upgrade the cluster to a new kubernetes version:
+
+    ```bash
+    ./bin/ck8s-kubespray run-playbook sc upgrade-cluster.yml -b --skip-tags=download
+    ./bin/ck8s-kubespray run-playbook wc upgrade-cluster.yml -b --skip-tags=download
+    ```
+
+1. If the environment is running `rook-ceph`, apply PSA labels to the namespace:
+
+    ```bash
+    kubectl label namespace rook-ceph pod-security.kubernetes.io/audit=privileged
+    kubectl label namespace rook-ceph pod-security.kubernetes.io/enforce=privileged
+    kubectl label namespace rook-ceph pod-security.kubernetes.io/warn=privileged
+    ```
```

Comment:
Do we need to add a step to also apply the new operator values? Or will the old PSPs be automatically removed?

Reply:
Should this be done before or after the upgrade? :thinking:

Reply:
Good question, I guess if we've enabled the PSA labels it will not break if we remove the PSPs before the upgrade. However I'd like to get this tested before we go with that

Reply:
I'm thinking these migration steps should actually be performed when migrating from 2.20 to 2.21, no matter what patch version you're migrating to, in addition to the [steps](https://github.com/elastisys/compliantkubernetes-kubespray/blob/v2.21.0-ck8s1/migration/v2.20.0-ck8sx-v2.21.0-ck8s1/upgrade-cluster.md) that are already there. The steps are not specifically tied to this PR and don't really make sense as I have added them now, as if you were migrating from `v2.21.0-ck8s1` to `v2.21.0-ck8s2` you would most likely already have run into issues if you didn't perform these steps when you upgraded to `v2.21.0-ck8s1` in the first place (In environments running `rook-ceph`).

Reply:
Yes, I totally agree. Let's add this to the 2.20 -> 2.21 upgrade instead